drivers/ethernet/eth_numaker.c

```diff
@@ -21,22 +21,33 @@
 
 #ifdef CONFIG_SOC_M467
 #include <m460_eth.h>
+#else
+#include <numaker_eth.h>
 #endif
 
 LOG_MODULE_REGISTER(eth_numaker, CONFIG_ETHERNET_LOG_LEVEL);
 
 /* Device EMAC Interface port */
 #define NUMAKER_GMAC_INTF  0
-/* 2KB Data Flash at 0xFF800 */
-#define NUMAKER_DATA_FLASH (0xFF800U)
+/* 2KB Data Flash */
+#define NUMAKER_DATA_FLASH DT_PROP_OR(DT_NODELABEL(emac), data_flash, 0xFF800U)
```

Comment:
Can you tell me where in the Technical Reference Manual this address is defined and said that that is for reading a mac address? I unfortunately was unsuccessful to find it.

Reply:
No TRM mentioned it.
Just to reserve one data flash for EMAC, system house could burn the specific mac-address in this data flash in factory stage.

Reply:
In zephyr we normally don't allow such undocumented behavior in ethernet drivers. It would be okay if it would be a documented reserved area by the soc manufacturer, but it isn't. In that case, it should be implemented in the app code by the user with using the api like ``int ret = net_mgmt(NET_REQUEST_ETHERNET_SET_MAC_ADDRESS, iface, &params, sizeof(struct ethernet_req_params));`` If I looked correctly in the TRM, I noticed that it was in the area for the application, which I don't thing, as there could be some other data there.

For the m55m1x series this behavior shouldn't be used at all, but for the m46x series if removal or or at least putting a Kconfig option around it, would be the best way.

Reply:
what do the other ethernet collaborators think about it @decsny @lmajewski @pdgendt  ?

Reply:
@maass-hamburg ,
Refer to other eth driver implementation, to support gen_random_mac() and removed emac data flash.
Thanks of your advice.

Reply:
good, maybe we should inform existing users via the migration guide that this functionality was removed and they aren't surprised

Reply:
hmm,
So far, no user asked how to burn mac-address in this specific data flash in zephyr platform.  Thus, in zephyr platform, there should be no developer to utilize this data flash yet.
Where is the migration guide to inform the change ?

Reply:
Users could have read the source code and figured it out themself.
in doc/releases/migration-guide-4.2.rst

Reply:
@maass-hamburg ,
Got it.
The requested change label is still on, is there any other concern ?

Reply:
you didn't pushed the change of the migration-guide

Reply:
Thanks and pushed migration-guide-4.2.rst